ext/web/lib.rs

```diff
@@ -314,46 +316,24 @@ impl Resource for TextDecoderResource {
   }
 }
 
-#[derive(Serialize)]
-#[serde(rename_all = "camelCase")]
-struct EncodeIntoResult {
-  read: usize,
-  written: usize,
-}
-
-#[op]
+#[op(v8)]
 fn op_encoding_encode_into(
-  input: String,
+  scope: &mut v8::HandleScope,
+  input: serde_v8::Value,
   buffer: &mut [u8],
-) -> EncodeIntoResult {
-  // Since `input` is already UTF-8, we can simply find the last UTF-8 code
-  // point boundary from input that fits in `buffer`, and copy the bytes up to
-  // that point.
-  let boundary = if buffer.len() >= input.len() {
-    input.len()
-  } else {
-    let mut boundary = buffer.len();
-
-    // The maximum length of a UTF-8 code point is 4 bytes.
-    for _ in 0..4 {
-      if input.is_char_boundary(boundary) {
-        break;
-      }
-      debug_assert!(boundary > 0);
-      boundary -= 1;
-    }
-
-    debug_assert!(input.is_char_boundary(boundary));
-    boundary
-  };
-
-  buffer[..boundary].copy_from_slice(input[..boundary].as_bytes());
-
-  EncodeIntoResult {
-    // The `read` output parameter is measured in UTF-16 code units.
-    read: input[..boundary].encode_utf16().count(),
-    written: boundary,
-  }
+  out_buf: &mut [u32],
+) {
+  let s = v8::Local::<v8::String>::try_from(input.v8_value).unwrap();
```

Comment:
Are you completely sure this is infallible?

Reply:
fixed